src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2247,15 +2247,23 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 									  &queryHavingQual, &queryTargetList,
 									  &queryGroupClause);
 
-	if (extendedOpNodeProperties->onlyPushableWindowFunctions)
-	{
-		ProcessWindowFunctionsForWorkerQuery(originalWindowClause,
-											 originalTargetEntryList,
-											 &queryWindowClause, &queryTargetList);
-	}
-	else
+	/*
+	 * Planner optimizations may leave window clauses with hasWindowFuncs as false.
+	 * Ignore window clauses in that case.
+	 */
+	if (extendedOpNodeProperties->hasWindowFuncs)
```

Comment:
TODO: check what happens when one has a WINDOW clause which is only referenced by window functions in subqueries. We should be erroring out in that scenario if postgres doesn't already